core/src/main/java/io/grpc/internal/RetriableStream.java

```diff
@@ -173,9 +194,29 @@ public ClientStreamTracer newClientStreamTracer(CallOptions callOptions, Metadat
     Metadata newHeaders = updateHeaders(headers, previousAttempts);
     // NOTICE: This set _must_ be done before stream.start() and it actually is.
     sub.stream = newSubstream(tracerFactory, newHeaders);
+    if (isHedging()) {
+      synchronized (lock) {
+        activeHedges.add(sub);
+      }
+    }
     return sub;
   }
 
+  private boolean isHedging() {
```

Comment:
This is a pretty scary method. 1) It has the same name as a variable that it initializes, but it isn't clear when to use the variable vs this method. 2) If called twice, it runs everything again, potentially blindly overwriting activeHedges. Since this is called from createSubstream, I think it is broken. 3) It's non-obvious that asking for the hedging policy sets the retry policy.